src/iam/iam-base.ts

```diff
@@ -306,17 +306,25 @@ export class IAMBase {
     }
   }
 
-  private async getPublicKey() {
+  private async getPublicKey(): Promise<string> {
     if (this._signer) {
       const address = await this._signer.getAddress();
       const hash = keccak256(address);
+      const sig = await this._signer.signMessage(arrayify(hash));
+
+      // Computation of the digest in order to recover the public key under the
+      // assumption that signer was made as per the eth_sign spec (https://eth.wiki/json-rpc/API#eth_sign)
       const digest = hashMessage(arrayify(hash));
-      if (((this._signer?.provider as any)?.provider as WalletConnectProvider)?.isWalletConnect) {
-        const sig = await this._signer.signMessage(arrayify(digest));
-        return computePublicKey(recoverPublicKey(digest, sig), true).slice(2);
+      const publicKey_digest = recoverPublicKey(digest, sig);
+      if (address === computeAddress(publicKey_digest)) {
```

Comment:
You might run into issues comparing non-checksum with checksum addresses. See https://docs.ethers.io/v4/notes.html#checksum-address.

```suggestion
      if (getAddress(address) === getAddress(computeAddress(publicKey_digest))) {
```

Reply:
Thanks, I added `getAddress`